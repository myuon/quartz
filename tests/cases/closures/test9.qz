fun main() {
    let a = 10;

    let f = fun () {
        println("{}", a.to_string());
    };

    println("{}", a.to_string());

    f();

    println("ok");
}

