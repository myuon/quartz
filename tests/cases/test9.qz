struct Point {
    x: i32,
    y: i32,
    z: i32,
}

fun main(): i32 {
    let p = Point {
        x: 10,
        y: 20,
        z: 0,
    };
    p.z = p.x + p.y;

    return p.z;
}

