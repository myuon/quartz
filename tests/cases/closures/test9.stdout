10
10
ok
